lib/config/fileconf.go

```diff
@@ -174,82 +186,152 @@ func MakeSampleFileConfig(flags SampleFlags) (fc *FileConfig, err error) {
+		if flags.Version == defaults.TeleportConfigVersionV2 {
+			a.ProxyListenerMode = types.ProxyListenerMode_Multiplex
+		}
+	} else {
+		a.EnabledFlag = "no"
 	}
 
 	// sample proxy config:
 	var p Proxy
-	p.EnabledFlag = "yes"
-	p.ListenAddress = conf.Proxy.SSHAddr.Addr
-	if flags.ACMEEnabled {
-		p.ACME.EnabledFlag = "yes"
-		p.ACME.Email = flags.ACMEEmail
-		// ACME uses TLS-ALPN-01 challenge that requires port 443
-		// https://letsencrypt.org/docs/challenge-types/#tls-alpn-01
-		p.PublicAddr = apiutils.Strings{net.JoinHostPort(flags.ClusterName, fmt.Sprintf("%d", teleport.StandardHTTPSPort))}
-		p.WebAddr = net.JoinHostPort(defaults.BindIP, fmt.Sprintf("%d", teleport.StandardHTTPSPort))
-	}
-	if flags.PublicAddr != "" {
-		// default to 443 if port is not specified
-		publicAddr, err := utils.ParseHostPortAddr(flags.PublicAddr, teleport.StandardHTTPSPort)
-		if err != nil {
-			return nil, trace.Wrap(err)
+	if roles[defaults.RoleProxy] {
+		p.EnabledFlag = "yes"
+		p.ListenAddress = conf.Proxy.SSHAddr.Addr
+		if flags.ACMEEnabled {
+			p.ACME.EnabledFlag = "yes"
+			p.ACME.Email = flags.ACMEEmail
+			// ACME uses TLS-ALPN-01 challenge that requires port 443
+			// https://letsencrypt.org/docs/challenge-types/#tls-alpn-01
+			p.PublicAddr = apiutils.Strings{net.JoinHostPort(flags.ClusterName, fmt.Sprintf("%d", teleport.StandardHTTPSPort))}
+			p.WebAddr = net.JoinHostPort(defaults.BindIP, fmt.Sprintf("%d", teleport.StandardHTTPSPort))
 		}
-		p.PublicAddr = apiutils.Strings{publicAddr.String()}
+		if flags.PublicAddr != "" {
+			// default to 443 if port is not specified
+			publicAddr, err := utils.ParseHostPortAddr(flags.PublicAddr, teleport.StandardHTTPSPort)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			p.PublicAddr = apiutils.Strings{publicAddr.String()}
+
+			// use same port for web addr
+			webPort := publicAddr.Port(teleport.StandardHTTPSPort)
+			p.WebAddr = net.JoinHostPort(defaults.BindIP, fmt.Sprintf("%d", webPort))
+		}
+		if flags.KeyFile != "" && flags.CertFile != "" {
+			if _, err := tls.LoadX509KeyPair(flags.CertFile, flags.KeyFile); err != nil {
+				return nil, trace.Wrap(err, "failed to load x509 key pair from --key-file and --cert-file")
+			}
+
+			p.KeyPairs = append(p.KeyPairs, KeyPair{
+				PrivateKey:  flags.KeyFile,
+				Certificate: flags.CertFile,
+			})
+		}
+	} else {
+		p.EnabledFlag = "no"
+	}
+
+	// Apps config:
+	var apps Apps
+	// assume users want app role if they added app name and/or uri but didn't add app role
+	if roles[defaults.RoleApp] || flags.AppURI != "" || flags.AppName != "" {
+		apps.EnabledFlag = "yes"
+		apps.Apps = []*App{
+			{
+				Name: flags.AppName,
+				URI:  flags.AppURI,
+			},
+		}
+	}
 
-		// use same port for web addr
-		webPort := publicAddr.Port(teleport.StandardHTTPSPort)
-		p.WebAddr = net.JoinHostPort(defaults.BindIP, fmt.Sprintf("%d", webPort))
+	// DB config:
+	var dbs Databases
+	if roles[defaults.RoleDatabase] {
```

Comment:
Is there any value in enabling the database or desktop services without actually including any databases/desktops?

Reply:
I don't think there is much value in it, but I think selecting those roles and ending up with a config file without their sections is worse.

Reply:
For now the focus is on improving the getting started process for SSH servers.

I would be fine if `teleport configure` failed if you passed database or desktop to the `--role` flag. Alternatively, write a warning to stderr: "role windowsdesktop requires manual configuration, windows_desktop_service will be disabled"`

Reply:
Added on commit https://github.com/gravitational/teleport/pull/11766/commits/ae31da5e1434081b097171b99ad045be11865653

```bash
$ ./build/teleport configure --roles=proxy,db,windowsdesktop
...
ould not check the contents of /var/lib/teleport: open /var/lib/teleport: permission denied
The data directory may contain existing cluster state.
Role db requires further configuration, db_service will be disabled
Role windowsdesktop requires further configuration, windows_desktop_service will be disabled
```